smartsim/generation/generator.py

```diff
@@ -80,12 +80,13 @@ def generate_experiment(self, *args):
         specified with a tag within the input file itself.
         The default tag is surronding an input value with semicolons.
         e.g. ``THERMO=;90;``
+
         """
-        entities, entity_lists, orchestrator = separate_entities(args)
+        generator_manifest = Manifest(*args)
         self._gen_exp_dir()
-        self._gen_orc_dir(orchestrator)
-        self._gen_entity_list_dir(entity_lists)
-        self._gen_entity_dirs(entities)
+        self._gen_orc_dir(generator_manifest.db)
```

Comment:
Should we just pass the ``Manifest`` object itself?

Reply:
OK, I like the idea of unifying the interface.

Reply:
@Spartee I'm actually going to push back on this. `_gen_entity_dirs` is used in a slightly different way (because it also has an optional parameter) and passing a `Manifest` to it would generate duplicates.

Reply:
Should we edit the generator code then to make the interface cleaner? My guess is we could make some of the code more re-usable? Maybe not in this ticket but file a future one? Im going to leave that up to you.